### PR TITLE
[PRED-2817] Fix failing Python3.5 tests and a bit of cleanup

### PR DESCRIPTION
**Pull request by dotkas**
# BATCH SCORING PULL REQUEST

This is a pull request into a **public** repository for Batch Scoring script maintained by DataRobot.

## RATIONALE

There was a bug where docker-compose would fail on Python 3.5 tests, leaving developers with a bad night's sleep.

## CHANGES

The failing tests seemed to be caused by overuse of statefulness in the docker-compose files. That's bad practice. However, as we are shifting focus to the Python API Client + Batch Scoring, I will not spend too much time on this.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,5 +72,7 @@ clean:
 	@rm -rf datarobot_batch_scoring.egg-info build/* dist/* 
 	@rm -rf htmlcov
 	@rm -rf .coverage
+	@rm -rf .cache
 	@rm -f batch_scoring.spec batch_scoring_sse.spec batch_scoring_deployment_aware.spec
+	@rm -f *.shelve.*
```
Comment (andrii-rubtsov) [inline Makefile]: Just curious what is in there?

File: docker-compose.yml [hunk 2]
```diff
@@ -7,35 +7,13 @@ services:
...
-        dockerfile: Dockerfile-3.3
-     volumes:
-        - .:/opt/project
-  python34:
-     image: batch-scoring-3.4
-     build:
-        context: .
-        dockerfile: Dockerfile-3.4
-     volumes:
-        - .:/opt/project
```
Comment (andrii-rubtsov) [inline docker-compose.yml]: In our README we state that we support all python 3+ versions. I think it is time to update it to 3.5+.
Maybe as a separate PR

File: Makefile [hunk 3]
```diff
@@ -18,7 +18,7 @@ checkrst:
 	# page properly.
 	python setup.py check --restructuredtext
 
-test: .install .install-test-deps flake8
+test: clean .install .install-test-deps flake8
```
Comment (andrii-rubtsov) [inline Makefile]: I assume this is the main contribution to fixing the problem, right?

Comment (dotkas) [reply Makefile]: Yes exactly.

File: Makefile (hunk 1 above)
Comment (dotkas) [reply Makefile]: As far as I understand it is sort of like Pickle files being made by the module during runtime - it was already being git-ignored but not removed during cleaning, if I understand it correctly. https://docs.python.org/3/library/shelve.html

File: docker-compose.yml (hunk 2 above)
Comment (dotkas) [reply docker-compose.yml]: I agree 👍

**Pull request merged**